### PR TITLE
feat: add osmosis amino converters

### PR DESCRIPTION
**Pull request by pastaghost**
* adds amino converters for Osmosis staking, lp, and bonding messages

### CODE REVIEW

**Review by elmutt — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -31,12 +31,15 @@
     "typescript": "^4.2.4"
   },
   "dependencies": {
+    "@cosmjs/amino": "^0.29.4",
+    "@cosmjs/crypto": "0.28.13",
     "@cosmjs/encoding": "^0.28.13",
     "@cosmjs/proto-signing": "^0.28.13",
     "@cosmjs/stargate": "^0.28.13",
     "@cosmjs/tendermint-rpc": "^0.28.13",
+    "@osmonauts/lcd": "^0.8.0",
```
Comment (kaladinlight) [inline package.json]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -31,12 +31,15 @@
     "typescript": "^4.2.4"
   },
   "dependencies": {
+    "@cosmjs/amino": "^0.29.4",
+    "@cosmjs/crypto": "0.28.13",
     "@cosmjs/encoding": "^0.28.13",
     "@cosmjs/proto-signing": "^0.28.13",
     "@cosmjs/stargate": "^0.28.13",
     "@cosmjs/tendermint-rpc": "^0.28.13",
```
Comment (kaladinlight) [inline package.json]: as long as you are updating dependencies, i would think we would want to keep them all on the same version 0.29.4. we also don't use `tendermint-rpc` explicitly so can remove

File: package.json [hunk 3]
```diff
@@ -31,12 +31,15 @@
...
+    "@osmonauts/lcd": "^0.8.0",
     "cosmjs-types": "^0.5.1",
     "google-protobuf": "^3.17.0",
-    "long": "^4.0.0"
+    "osmojs": "^0.36.0"
```
Comment (kaladinlight) [inline package.json]: `0.37.0` is most recent if we want to use that

File: src/amino/osmosis/index.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { AminoConverters } from '@cosmjs/stargate';
+import { osmosisAminoConverters } from 'osmojs'
+
+
```
Comment (kaladinlight) [inline src/amino/osmosis/index.ts]:
```suggestion
```

File: src/index.ts [hunk 5]
```diff
@@ -12,12 +12,14 @@ import {
   createIbcAminoConverters,
   createStakingAminoConverters,
 } from '@cosmjs/stargate'
+
+
```
Comment (kaladinlight) [inline src/index.ts]:
```suggestion
```

File: src/index.ts [hunk 6]
```diff
@@ -47,6 +49,8 @@ export async function sign(
   authInfoBytes: string
   signatures: string[]
 }> {
+
+
```
Comment (kaladinlight) [inline src/index.ts]:
```suggestion
```

File: package.json [hunk 7]
```diff
@@ -31,12 +31,15 @@
...
+    "@cosmjs/amino": "^0.29.4",
+    "@cosmjs/crypto": "0.28.13",
     "@cosmjs/encoding": "^0.28.13",
     "@cosmjs/proto-signing": "^0.28.13",
     "@cosmjs/stargate": "^0.28.13",
     "@cosmjs/tendermint-rpc": "^0.28.13",
+    "@osmonauts/lcd": "^0.8.0",
     "cosmjs-types": "^0.5.1",
     "google-protobuf": "^3.17.0",
-    "long": "^4.0.0"
```
Comment (kaladinlight) [inline package.json:40]: we are importing long in `src/proto/index.tx` and should keep explicit import. version 5.2.1 is current if we want to update

**Comment by kaladinlight** [comment]
The only other outstanding comment would be that we have the `osmosis-frontend` submodule currently to provide protobuf types for osmosis, but these are now provided by osmojs. As a cleanup, it would be nice to remove the submodule and update the osmo protobuf types to use osmojs.

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**